pkg/render/intrusiondetection/dpi/dpi.go

```diff
@@ -180,6 +183,25 @@ func (d *dpiComponent) dpiDaemonset() *appsv1.DaemonSet {
 	if d.cfg.TyphaNodeTLS.NodeSecret.UseCertificateManagement() {
 		initContainers = append(initContainers, d.cfg.TyphaNodeTLS.NodeSecret.InitContainer(DeepPacketInspectionNamespace))
 	}
+	if d.cfg.IntrusionDetection.Spec.DeepPacketInspectionDaemonset.SnortInitContainers != nil {
+		for iter, snortInitContainer := range d.cfg.IntrusionDetection.Spec.DeepPacketInspectionDaemonset.SnortInitContainers {
+			container := corev1.Container{
+				Name:            fmt.Sprintf("%s-%d", DeepPacketInspectionInitContainerName, iter),
+				Image:           snortInitContainer.Image,
+				ImagePullPolicy: corev1.PullAlways,
+				VolumeMounts: []corev1.VolumeMount{
+					{
+						Name:      DeepPacketInspectionSnortRulesVolumeName,
+						MountPath: DeepPacketInspectionSnortRulesVolumePath,
+					},
+				},
+			}
+			if snortInitContainer.Resources != nil {
```

Comment:
Instead of doing this explicitly, we should follow the established "overrides" pattern: https://github.com/tigera/operator/blob/dcb21dfb148f47b5dfd39bd136398e04c4c21935/pkg/render/intrusion_detection.go#L538-L540

Reply:
There is an equivalent function for DaemonSet: https://github.com/tigera/operator/blob/dcb21dfb148f47b5dfd39bd136398e04c4c21935/pkg/render/common/components/components.go#L103-L126

You would just need to make your struct implement this interface: https://github.com/tigera/operator/blob/dcb21dfb148f47b5dfd39bd136398e04c4c21935/pkg/components/overrides.go#L24-L60

Most functions will return `nil` to indicate that we don't support overrides on them, but the `GetInitContainers` will merge in the resources appropriately.

Reply:
We agreed this is OK.